src/commutator_rpnl.F

```diff
@@ -398,4 +402,509 @@ SUBROUTINE build_com_rpnl(matrix_rv, qs_kind_set, sab_orb, sap_ppnl, eps_ppnl)
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'build_com_mom_nl', &
+         routineP = moduleN//':'//routineN
+
+      INTEGER :: handle, i, iab, iac, iatom, ibc, icol, ikind, ilist, ind, inode, irow, iset, &
+         jatom, jkind, jneighbor, kac, katom, kbc, kkind, l, lc_max, lc_min, ldai, ldsab, lppnl, &
+         maxco, maxder, maxl, maxlgto, maxlppnl, maxppnl, maxsgf, mepos, na, nb, ncoa, ncoc, &
+         nkind, nlist, nneighbor, nnode, np, nppnl, nprjc, nseta, nsgfa, nthread, order, prjc, sgfa
+      INTEGER, DIMENSION(3)                              :: cell_b, cell_c
+      INTEGER, DIMENSION(:), POINTER                     :: la_max, la_min, npgfa, nprj_ppnl, &
+                                                            nsgf_seta
+      INTEGER, DIMENSION(:, :), POINTER                  :: first_sgfa
+      LOGICAL                                            :: asso_rrv, asso_rv, asso_rxrv, found, go, &
+                                                            gpot, my_ref, my_rrv, my_rv, my_rxrv, &
+                                                            ppnl_present, spot
+      REAL(KIND=dp)                                      :: dac, ppnl_radius
+      REAL(KIND=dp), ALLOCATABLE, DIMENSION(:, :, :)     :: ai_work, sab, work
+      REAL(KIND=dp), DIMENSION(1)                        :: rprjc, zetc
+      REAL(KIND=dp), DIMENSION(3)                        :: ra, rab, rac, raf, rc, rcf, rf
+      REAL(KIND=dp), DIMENSION(:), POINTER               :: alpha_ppnl, set_radius_a
+      REAL(KIND=dp), DIMENSION(:, :), POINTER            :: cprj, rpgfa, sphi_a, vprj_ppnl, zeta
+      REAL(KIND=dp), DIMENSION(:, :, :), POINTER         :: achint, acint, bchint, bcint
+      TYPE(alist_type), POINTER                          :: alist_ac, alist_bc
+      TYPE(block_p_type), DIMENSION(:), POINTER          :: blocks_rrv, blocks_rv, blocks_rxrv
+      TYPE(clist_type), POINTER                          :: clist
+      TYPE(gth_potential_p_type), DIMENSION(:), POINTER  :: gpotential
+      TYPE(gth_potential_type), POINTER                  :: gth_potential
+      TYPE(gto_basis_set_p_type), DIMENSION(:), POINTER  :: basis_set
+      TYPE(gto_basis_set_type), POINTER                  :: orb_basis_set
+      TYPE(neighbor_list_iterator_p_type), &
+         DIMENSION(:), POINTER                           :: nl_iterator
+      TYPE(sap_int_type), DIMENSION(:), POINTER          :: sap_int
+      TYPE(sgp_potential_p_type), DIMENSION(:), POINTER  :: spotential
+      TYPE(sgp_potential_type), POINTER                  :: sgp_potential
+
+      CALL timeset(routineN, handle)
+
+      my_rxrv = .FALSE.
+      my_rrv = .FALSE.
+      my_rv = .FALSE.
+      IF (PRESENT(matrix_rxrv)) my_rxrv = .TRUE.
+      IF (PRESENT(matrix_rrv)) my_rrv = .TRUE.
+      IF (PRESENT(matrix_rv)) my_rv = .TRUE.
+      IF (.NOT. (my_rv .OR. my_rxrv .OR. my_rrv)) RETURN
+
+      ppnl_present = ASSOCIATED(sap_ppnl)
+
+      IF (ppnl_present) THEN
```

Comment:
since you already have early exits I'd also use the same pattern here and do `IF (.NOT. ppnl_present) RETURN`